sql/type.go

```diff
@@ -386,6 +386,10 @@ func (t numberT) Compare(a interface{}, b interface{}) (int, error) {
 func (t numberT) String() string { return t.t.String() }
 
 func compareFloats(a interface{}, b interface{}) (int, error) {
+	if hasNulls, res := hasNullInputs(a, b); hasNulls {
```

Comment:
I would break this null sorting change into its own PR and feed it back to them separately, since it's independent of the new update feature.

Reply:
Moving this to its own PR would cause the related tests in this PR to fail. Is that okay? This change is fairly core to the correct behavior of `Update` working as one would expect.